datafusion/sql/src/select.rs

```diff
@@ -778,9 +777,12 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         // Rewrite the HAVING expression to use the columns produced by the
         // aggregation.
         let having_expr_post_aggr = if let Some(having_expr) = having_expr_opt {
-            let having_expr_post_aggr =
-                rebase_expr(having_expr, &aggr_projection_exprs, input)?;
-
+            let having_expr_post_aggr = match having_expr {
+                Expr::Literal(ScalarValue::Boolean(Some(false))) => {
+                    having_expr.to_owned()
```

Comment:
Can it fix for other constant expressions 🤔, such as
```sql
SELECT AVG(v1) FROM t1 GROUP BY 1 != 1 having 1 != 1;
```
Looks like this is also an issue related to PushdownFilter.

Reply:
Yes, I can fix other test coverage